File: task2_python/factorial.py
def factorial(n: int) -> int:
    if n < 0:
        raise ValueError("Число должно быть неотрицательным")
    if n == 0:
        return 1
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result


# ── Точка входа ──────────────────────────────────────────────
if __name__ == "__main__":
    test_values = [0, 1, 5, 6, 10]
    print("=" * 40)
    print("  Факториал (Python)")
    print("=" * 40)
    for number in test_values:
        print(f"  factorial({number}) = {factorial(number)}")

File: task2_java/Factorial.java
public class Factorial {

    public static long factorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException(
                "Число должно быть неотрицательным"
            );
        }
        if (n == 0) {
            return 1;
        }
        long result = 1;
        for (int i = 1; i <= n; i++) {
            result *= i;
        }
        return result;
    }

    public static void main(String[] args) {
        int[] testValues = {0, 1, 5, 6, 10};
        System.out.println("========================================");
        System.out.println("  Факториал (Java)");
        System.out.println("========================================");
        for (int number : testValues) {
            System.out.println(
                "  factorial(" + number + ") = " + factorial(number)
            );
        }
    }
}
